### ISSUE TITLE
bug: copy fails silently in Safari

### ISSUE THREAD

**Issue by jimniels** [quadratichq/quadratic#2018]
In safari, if I copy (CMD+C ) from a cell, then paste in another, sometimes the copy operation just fails and it pastes whatever was on my clipboard before I came to the app.

I honestly don't know how to get copy to work again, which is one problem.

But also it would be good if when I hit "cmd+c" it popped up a toast or something that said "failed to copy to clipboard". I'm not sure how safari users remediate this issue? Is it a permissions thing?

https://github.com/user-attachments/assets/df92332c-303a-40ea-9f68-6b727c032eb0

**Comment by anthonycorletti** [comment]
yeah... safari is picky. https://stackoverflow.com/questions/62327358/javascript-clipboard-api-safari-ios-notallowederror-message. here's what im seeing:

<img width="992" alt="Image" src="https://github.com/user-attachments/assets/8a92b9c0-3c33-499e-88ed-f8e13d40dbdc" />

looks like its happening here for me https://github.com/quadratichq/quadratic/blob/qa/quadratic-client/src/app/grid/actions/clipboard/clipboard.ts#L114

**Comment by davidkircos** [comment]
Thanks @anthonycorletti , we will investigate. It's possible that Safari changed security requirements since our implementation

**Comment by davidfig** [comment]
We should try to fix this. Safari will be useful once we have mobile support. Few people use chrome on iPhones.
